test/integration/crud/maxTimeMS.test.ts

```diff
@@ -0,0 +1,268 @@
+      client.connect(function () {
+        const db = client.db(configuration.db);
+        const col = db.collection('max_time_ms_2');
+
+        // Insert a couple of docs
+        const docs_1 = [{ agg_pipe: 1 }];
+
+        // Simple insert
+        col.insertMany(docs_1, { writeConcern: { w: 1 } }, function (err) {
+          expect(err).to.not.exist;
+
+          // Execute a find command
+          col
+            .find({ $where: 'sleep(100) || true' })
+            .maxTimeMS(50)
+            .toArray(function (err) {
+              test.ok(err != null);
+              client.close(done);
+            });
+        });
+      });
+    }
+  });
+
+  it('should correctly fail with maxTimeMS error', {
+    // Add a tag that our runner can trigger on
+    // in this case we are setting that node needs to be higher than 0.10.X to run
+    metadata: {
+      requires: {
+        topology: ['single', 'replicaset']
+      }
+    },
+
+    test: function (done) {
+      const configuration = this.configuration;
+      const client = configuration.newClient(configuration.writeConcernMax(), { maxPoolSize: 1 });
+      client.connect(function () {
+        const db = client.db(configuration.db);
+        const col = db.collection('max_time_ms_5');
+
+        // Insert a couple of docs
+        const docs_1 = [{ agg_pipe: 10 }];
+
+        // Simple insert
+        col.insertMany(docs_1, { writeConcern: { w: 1 } }, function (err) {
+          expect(err).to.not.exist;
+
+          db.admin().command(
+            { configureFailPoint: 'maxTimeAlwaysTimeOut', mode: 'alwaysOn' },
+            function (err, result) {
+              expect(err).to.not.exist;
+              test.equal(1, result?.ok);
+
+              col
+                .find({})
+                .maxTimeMS(10)
+                .toArray(function (err) {
+                  test.ok(err != null);
+
+                  db.admin().command(
+                    { configureFailPoint: 'maxTimeAlwaysTimeOut', mode: 'off' },
+                    function (err, result) {
+                      expect(err).to.not.exist;
+                      test.equal(1, result?.ok);
+                      client.close(done);
+                    }
+                  );
+                });
+            }
+          );
+        });
+      });
+    }
+  });
+
+  describe('awaitData, tailable, maxTimeMS, and maxAwaitTimeMS on cursors', () => {
+    const insertedDocs = [{ _id: 1 }];
+    let client: MongoClient;
+    let cappedCollection: Collection<{ _id: number }>;
+    let cursor: FindCursor<{ _id: number }>;
+    let events: CommandStartedEvent[];
+
+    beforeEach(async function () {
+      client = this.configuration.newClient({ monitorCommands: true });
+      await client
+        .db()
+        .dropCollection('cappedAt3')
+        .catch(() => null);
+      await sleep(100); // Need to make sure it has had time to drop entirely
```

Comment:
Why?  Does dropCollection return eagerly?

Reply:
I shouldn't need this, in sharded cluster topologies reusing a namespace that was just dropped isn't reccomended IIRC and I thought I was running into that. dropped.